src/App.test.js

```diff
@@ -1,8 +1,58 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import Operate from './logic/operate';
+import Calculate from './logic/calculate';
 
```

Comment:
Please add snapshots for all the components using Jest as it's a requirement. Moreover, please use `React Testing Library` to simulate user interaction.